src/Features/Core/Portable/Completion/CompletionService.cs

```diff
@@ -206,44 +207,99 @@ public virtual TextSpan GetDefaultCompletionListSpan(SourceText text, int caretP
             var itemsWithPatternMatch = items.SelectAsArray(
                 item => (item, completionHelper.GetMatch(item.FilterText, filterText, includeMatchSpans: false, CultureInfo.CurrentCulture)));
 
-            return FilterItems(completionHelper, itemsWithPatternMatch);
+            return FilterItems(completionHelper, itemsWithPatternMatch, filterText);
         }
 
         internal static ImmutableArray<CompletionItem> FilterItems(
             CompletionHelper completionHelper,
-            ImmutableArray<(CompletionItem item, PatternMatch? match)> itemsWithPatternMatch)
+            ImmutableArray<(CompletionItem item, PatternMatch? match)> itemsWithPatternMatch,
+            string filterText)
         {
-            var bestItems = ArrayBuilder<(CompletionItem, PatternMatch?)>.GetInstance();
-            foreach (var pair in itemsWithPatternMatch)
+            // It's very common for people to type expecting completion to fix up their casing,
+            // so if only lowercase characters were typed so far, we'd loosen our standard on comparing items
+            // in case-sensitive manner and take into consideration the MatchPriority as well.
+            // i.e. when everything else is equal, then if item1 is a better case-sensitive match but item2 has higher 
+            // MatchPriority, we consider them equally good match, so the controller will later have a chance to
+            // decide which is the best one to select.
+            var isFilterTextLowercaseOnly = !filterText.Any(char.IsUpper);
```

Comment:
slightly odd phrasing here.  given the variable name, i would expect this to be `filterText.All(char.IsLower)`.  if you're really doing `! ... Any(IsUpper)` then variable should be named `filterTextContainsNoUpperLetters` (or somethingl ike that).

Also, please avoid the linq call here.